kolibri/core/assets/src/views/FocusTrap.vue

```diff
@@ -59,15 +59,27 @@
         e.stopPropagation();
         if (!this.isTrapActive) {
           // On first focus, redirect to first option, then activate trap
-          this.firstEl.focus();
+          this.focusFirstEl();
           this.isTrapActive = true;
         } else {
-          this.lastEl.focus();
+          this.focusLastEl();
         }
       },
       handleLastTrapFocus(e) {
         e.stopPropagation();
-        this.firstEl.focus();
+        this.focusFirstEl();
+      },
+      focusFirstEl() {
+        this.$emit('shouldFocusFirstEl');
+        if (this.firstEl) {
+          this.firstEl.focus();
```

Comment:
Emitting `shouldFocusFirstEl`/`shouldFocusLastEl` events and shortly after focusing on the first/last element here in the `FocusTrap` seems to be confusing, as well as looking at `FocusTrap` API from outside:

```
<FocusTrap
  :firstEl="firstEl"
  :lastEl="lastEl"
  @shouldFocusFirstEl="firstEl.focus()"
  @shouldFocusLastEl="lastEl.focus()"
>
```
where it's not clear whether it's `FocusTrap` or its parent component that is responsible for handling focus.

One way to solve this would be to document in comments that when `firstEl`/`lastEl`  props are provided, `FocusTrap` will handle focus and `shouldFocusFirstEl`/`shouldFocusLastEl` events should be ignored.

However, I think it may be much simpler to remove this logic and `firstEl`/`lastEl` prop from `FocusTrap` API completely now that we can use events. This would also help us to have a consistent way of handling focuses everywhere. I can't imagine a scenario when we couldn't use events.

@bjester Do you have any thoughts?

Reply:
I am not @bjester, but that makes sense to me, @MisRob :)

Reply:
That works. Does it mean that we should change the `FocusTrap` API in this PR and update the other uses of `FocusTrap`, or create a new PR that does that and updates any additional uses of `FocusTrap`?

Reply:
I assume it'd make sense to have these changes logically together as sometimes happens in PRs when updating components APIs and wouldn't expect complications with updating old places, but maybe you can decide according to how it goes. I feel fine about both ways as long as it's addressed.

Reply:
Since this is a bug fix, and because the issue resided outside the `<FocusTrap>` implementation, I think it's @sairina's call as to whether she wants to close this out quickly or continue with additional refactoring. The situation definitely felt like it necessitated a new API for `<FocusTrap>`, so I think as a team, in this type of situation, we should strive for iterative development to avoid getting ourselves into larger tangential refactors. Long term, we should definitely replace the props with events as @MisRob said!

Reply:
Makes sense to me. Initially, I assumed that it wouldn't be a larger diff, but it is as I can see now. In any case, thanks @sairina for iterating on this all.